engine/Shopware/Components/Translation.php

```diff
@@ -391,6 +396,295 @@ public function write($language, $type, $key = 1, $data = null, $merge = false)
+     * and translates them.
+     *
+     * @param $orders
+     * @param $context
+     * @param $language integer Language id
+     * @param $fallback integer Id of fallback language
+     *
+     * @return mixed
+     */
+    public function translateOrders($orders, $context, $language = null, $fallback = null)
+    {
+        $documentTypes = [];
+        $paymentTypes = [];
+        $dispatchTypes = [];
+
+        // Extract documents, payment and dispatch methods
+        for ($i = 0; $i < count($orders); $i += 1) {
+            $orderDocuments = $orders[$i]['documents'];
+            $dispatchTypes[$orders[$i]['dispatch']['id']] = $orders[$i]['dispatch'];
+            $paymentTypes[$orders[$i]['payment']['id']] = $orders[$i]['payment'];
+            for ($orderCounter = 0; $orderCounter < count($orderDocuments); $orderCounter += 1) {
+                $type = $orderDocuments[$orderCounter]['type'];
+                $documentTypes[$type['id']] = $type;
+            }
+        }
+
+        // Translate the objects
+        $translatedDocumentTypes = $this->translateDocuments($documentTypes, $context, $language, $fallback);
+        $translatedDispatchTypes = $this->translateDispatches($dispatchTypes, $context, $language, $fallback);
+        $translatedPaymentTypes = $this->translatePayments($paymentTypes, $context, $language, $fallback);
+
+        // Save the translated objects
+        for ($i = 0; $i < count($orders); $i += 1) {
+            $orderDocuments = $orders[$i]['documents'];
+            for ($orderCounter = 0; $orderCounter < count($orderDocuments); $orderCounter += 1) {
+                $type = $orderDocuments[$orderCounter]['type'];
+                $orders[$i]['documents'][$orderCounter]['type'] = $translatedDocumentTypes[$type['id']];
+            }
+
+            if ($translatedDispatchTypes[$orders[$i]['dispatch']['id']]) {
+                $orders[$i]['dispatch'] = $translatedDispatchTypes[$orders[$i]['dispatch']['id']];
+            }
+
+            if ($translatedPaymentTypes[$orders[$i]['payment']['id']]) {
+                $orders[$i]['payment'] = $translatedPaymentTypes[$orders[$i]['payment']['id']];
+            }
+        }
+
+        return $orders;
+    }
+
+    /**
+     * Helper method to translate dispatches.
+     * Contains all configurations for dispatches and forwards the to the translateObjects method.
+     *
+     * @param $dispatches [] Objects
+     * @param $context Integer Calling context
+     * @param $language integer Language id
+     * @param $fallback integer Id of fallback language
+     *
+     * @return mixed Translated dispatches
+     */
+    public function translateDispatches($dispatches, $context, $language = null, $fallback = null)
+    {
+        $translationMapping = [['source' => 'dispatch_name', 'destination' => 'name']];
+
+        return $this->translateObjects(
+            $dispatches, 'config_dispatch', $context, $translationMapping, $language, $fallback
```

Comment:
Please either move all parameters to the first line or separate them to one parameter per line.